tilings/strategies/fusion/constructor.py

```diff
@@ -545,3 +548,188 @@ def random_sample_sub_objects(
+        # The right side of the fuse region contributes to the left-sided assumption
+        if right_fuse_param_idx is not None:
+            for param in left_sided_parameters:
+                param_idx_on_fuse = t_fuse.extra_parameters.index(
+                    extra_parameters[param]
+                )
+                res[right_fuse_param_idx].add(param_idx_on_fuse)
+        return tuple(map(tuple, res))
+
+    @staticmethod
+    def get_left_fuse_index(
+        left_sided_parameters: Tuple[str, ...],
+        fuse_parameter: str,
+        extra_parameters: Dict[str, str],
+        t_unfuse: Tiling,
+    ) -> Optional[int]:
+        """
+        Return the param index of the left sided param that fuses to the fuse param on
+        the fused tiling.
+        """
+        for parent_param, child_param in extra_parameters.items():
+            if child_param == fuse_parameter and parent_param in left_sided_parameters:
+                return t_unfuse.extra_parameters.index(parent_param)
+        return None
+
+    def forward_map(self, param: Parameters) -> Parameters:
+        """
+        Maps a set of parameters on the fuse tiling to a set of parameters on
+        the unfused tiling.
+        """
+        new_param = [0 for _ in range(self.num_fuse_param)]
+        assert len(param) == len(self.unfuse_pos_to_fuse_pos)
+        for pvalue, fuse_idxs in zip(param, self.unfuse_pos_to_fuse_pos):
+            for fuse_idx in fuse_idxs:
+                new_param[fuse_idx] += pvalue
+        return tuple(new_param)
+
+    def a_map(self, param: Parameters) -> Parameters:
+        if self.type == ReverseFusionConstructor.Type.BOTH:
+            # In the case where both is track we don't subtract anything
+            return tuple(-1 for _ in param)
+        if self.type == ReverseFusionConstructor.Type.LEFT_ONLY:
+            add = self.left_sided_index
+            substract = self.right_sided_index
+        else:
+            assert self.type == ReverseFusionConstructor.Type.RIGHT_ONLY
+            add = self.right_sided_index
+            substract = self.left_sided_index
+        new_param = list(param)
+        for idx in add:
+            new_param[idx] += 1
+        for idx in substract:
+            new_param[idx] -= 1
+        return tuple(new_param)
+
+    def get_terms(
+        self, parent_terms: Callable[[int], Terms], subterms: SubTerms, n: int
+    ) -> Terms:
+        terms: Terms = Counter()
+        child_terms = subterms[0](n)
+        for param, value in child_terms.items():
+            new_param = self.forward_map(param)
+            new_value = value - child_terms[self.a_map(param)]
+            assert new_param not in terms or new_value == terms[new_param]
+            terms[new_param] = new_value
+        return terms
+
+    def get_equation(
+        self, lhs_func: sympy.Function, rhs_funcs: Tuple[sympy.Function, ...]
+    ) -> sympy.Eq:
+        raise NotImplementedError
```

Comment:
Should we make this use the same method as the normal fusion constructor?

Reply:
Will make a fix on css to change the get equation of reverse rule to use the original rule